unit_test/tstUniformCartesianMeshMapping.hpp

```diff
@@ -0,0 +1,347 @@
+ * Copyright (c) 2021 by the Picasso authors                                *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Picasso library. Picasso is distributed under a *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Picasso_BatchedLinearAlgebra.hpp>
+#include <Picasso_CurvilinearMesh.hpp>
+#include <Picasso_Types.hpp>
+#include <Picasso_UniformCartesianMeshMapping.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cmath>
+
+#include <gtest/gtest.h>
+
+using namespace Picasso;
+
+namespace Test
+{
+//---------------------------------------------------------------------------//
+void mappingTest3d()
+{
+    // Global parameters.
+    Kokkos::Array<double, 6> global_box = { -10.0, -10.0, -10.0,
+                                            10.0,  10.0,  10.0 };
+    double cell_size = 0.5;
+    int num_cell = 20.0 / cell_size;
+    int halo_width = 1;
+    Kokkos::Array<bool, 3> periodic = { true, false, true };
+
+    // Partition only in the x direction.
+    std::array<int, 3> ranks_per_dim = { 1, 1, 1 };
+    MPI_Comm_size( MPI_COMM_WORLD, &ranks_per_dim[0] );
+
+    // Create the mesh and field manager.
+    auto manager = createUniformCartesianMesh( TEST_MEMSPACE{}, cell_size,
+                                               global_box, periodic, halo_width,
+                                               MPI_COMM_WORLD, ranks_per_dim );
+
+    // Check the mapping data.
+    auto mapping = manager->mesh().mapping();
+    using mapping_type = decltype( mapping );
+    EXPECT_EQ( cell_size, mapping._cell_size );
+    EXPECT_EQ( 1.0 / cell_size, mapping._inv_cell_size );
+    EXPECT_EQ( cell_size * cell_size * cell_size, mapping._cell_measure );
+    EXPECT_EQ( 1.0 / ( cell_size * cell_size * cell_size ),
+               mapping._inv_cell_measure );
+    for ( int d = 0; d < 3; ++d )
+    {
+        EXPECT_EQ( num_cell, mapping._global_num_cell[d] );
+        EXPECT_EQ( periodic[d], mapping._periodic[d] );
+    }
+
+    // Check grid.
+    const auto& global_grid = manager->mesh().localGrid()->globalGrid();
+    const auto& global_mesh = global_grid.globalMesh();
+
+    EXPECT_EQ( global_mesh.lowCorner( 0 ), 0.0 );
+    EXPECT_EQ( global_mesh.lowCorner( 1 ), 0.0 );
+    EXPECT_EQ( global_mesh.lowCorner( 2 ), 0.0 );
+
+    EXPECT_EQ( global_mesh.highCorner( 0 ), num_cell );
+    EXPECT_EQ( global_mesh.highCorner( 1 ), num_cell );
+    EXPECT_EQ( global_mesh.highCorner( 2 ), num_cell );
```

Comment:
10?

Reply:
The Cajita mesh is in the logical frame so it spans `(0.0,NumCell)` in each dimension